### PR TITLE
appengine: updating dependency paths and adding documentation

### PR DESCRIPTION
**Pull request by kenshaw**
Ran aefix against the gaedatastore and added some documentation to the
appengine folder along with an example for deploying to the traditional
and flexible environments.

### CODE REVIEW

**Comment by kenshaw** [comment]
Please review and let me know if this can be integrated. Thanks!

**Comment by kenshaw** [comment]
I had to I update the Glide dependencies, and also subsequently updated the go_appengine SDK used with travis-ci.

File: .travis.yml [hunk 1]
```diff
@@ -16,10 +16,10 @@ install:
 
   # Google App Engine dependencies
   - cd ..
-  - wget https://storage.googleapis.com/appengine-sdks/featured/go_appengine_sdk_linux_amd64-1.9.23.zip
-  - unzip -q go_appengine_sdk_linux_amd64-1.9.23.zip
+  - wget https://storage.googleapis.com/appengine-sdks/featured/go_appengine_sdk_linux_amd64-1.9.40.zip
+  - unzip -q go_appengine_sdk_linux_amd64-1.9.40.zip
   - export PATH=$PATH:$PWD/go_appengine/
-  - $PWD/go_appengine/goapp get github.com/stretchr/testify/require github.com/pborman/uuid golang.org/x/net/context # FIXME
+  #  - $PWD/go_appengine/goapp get github.com/stretchr/testify/require github.com/pborman/uuid golang.org/x/net/context
```
Comment (dennwc) [inline .travis.yml]: You may safely remove this line if it works and runs the tests without it.

File: appengine/app.yaml [hunk 2]
```diff
@@ -1,7 +1,5 @@
-application: cayley-test
-version: 1
 runtime: go
-api_version: go1.6.3
+api_version: go1
```
Comment (dennwc) [inline appengine/app.yaml]: Any reason to unpin it from 1.6.3?

**Comment by dennwc** [comment]
What is the `appengine/appengine` binary file attached?

File: appengine/app.yaml (hunk 2 above)
Comment (kenshaw) [inline appengine/app.yaml]: appengine only supports go1 in this field. see here: https://cloud.google.com/appengine/docs/go/config/appref

File: .travis.yml (hunk 1 above)
Comment (kenshaw) [inline .travis.yml]: Will do.

**Comment by kenshaw** [comment]
The appengine/appengine binary is attached because I'm an idiot. I will fix this.

**Comment by kenshaw** [comment]
I made the needed changes. Let me know if there is any other problems / issues. Again, thanks!

**Comment by dennwc** [comment]
LGTM

**Pull request merged**